### PR TITLE
Combine CommonMessageAttributes and CommonName

### PR DESCRIPTION
**Pull request by kaituo**
Note: since there are a lot of dependencies, I only list the main class and test code to save reviewers' time. The build will fail due to missing dependencies. I will use that PR just for review. will not merge it. Will have a big one in the end and merge once after all review PRs get approved. Now the code is missing unit tests. Will add unit tests, run performance tests, and fix bugs before the official release.

### Description
Both CommonMessageAttributes and CommonName define some key or name used as in Json keys or log messages.  This PR combines these two class by moving the content from CommonMessageAttributes to CommonName.  This is so we can maintain such keys or attributes in a central place.

Testing done:
1. Build passes.
 
### Check List
- [ Y ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/ad/AbstractADTest.java [hunk 1]
```diff
@@ -212,7 +216,7 @@ protected static void setUpThreadPool(String name) {
                 AnomalyDetectorPlugin.AD_THREAD_POOL_NAME,
                 1,
                 1000,
-                "opendistro.ad." + AnomalyDetectorPlugin.AD_THREAD_POOL_NAME
+                "opensearch.ad." + AnomalyDetectorPlugin.AD_THREAD_POOL_NAME
```
Comment (ohltyler) [inline src/test/java/org/opensearch/ad/AbstractADTest.java:219]: Nice find. Can you grep in your final codebase that has all of the changes, that there are no more references of `opendistro`?

Comment (kaituo) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:219]: There are quite a few of opendistro like

src/main/java/org/opensearch/ad/constant/CommonName.java
34:    public static final String CHECKPOINT_INDEX_NAME = ".opendistro-anomaly-checkpoints";
36:    public static final String DETECTION_STATE_INDEX = ".opendistro-anomaly-detection-state";
40:    public static final String ANOMALY_RESULT_INDEX_ALIAS = ".opendistro-anomaly-results";

According to infra team, our setting will have backward compatibility to opendistro, while our indices will use keep using opendistro.

Comment (ohltyler) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:219]: Understood, agreed the BWC vars and the indices will remain `opendistro`, was more referring to other places that can be changed like the thread pool name above.

File: src/test/java/org/opensearch/ad/AbstractADTest.java [hunk 2]
```diff
@@ -223,17 +227,33 @@ protected static void tearDownThreadPool() {
...
 
-    public void setupTestNodes(Settings settings, TransportInterceptor transportInterceptor) {
+    /**
+     *
+     * @param transportInterceptor Interceptor to for transport requests. Used
+     *  to mock transport layer.
+     * @param nodeSettings node override of setting
+     * @param setting the supported setting set.
+     */
+    public void setupTestNodes(TransportInterceptor transportInterceptor, final Settings nodeSettings, Setting<?>... setting) {
```
Comment (ohltyler) [inline src/test/java/org/opensearch/ad/AbstractADTest.java:237]: Can you explain why you've added the additional `Setting` set here to pass to `FakeNode`?

File: src/test/java/org/opensearch/ad/AbstractADTest.java (hunk 1 above)
Comment (kaituo) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:219]: right, fixed a few in our docs and comments.  Please check the new commit.

File: src/test/java/org/opensearch/ad/AbstractADTest.java (hunk 2 above)
Comment (kaituo) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:237]: Please see an example: https://github.com/kaituo/anomaly-detection-1/blob/multiCat4/src/test/java/org/opensearch/ad/transport/AnomalyResultTests.java#L437-L442

The added setting set is to list supported setting set.  It is used as part of testing setting changes.

Haven't sent out tests in PRs as I am still adding unit tests.

Comment (ohltyler) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:237]: Got it, makes sense.

File: src/test/java/org/opensearch/ad/AbstractADTest.java (hunk 1 above)
Comment (ohltyler) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:219]: Not sure what commit you are referring to?

Comment (kaituo) [reply src/test/java/org/opensearch/ad/AbstractADTest.java:219]: this commit: https://github.com/opensearch-project/anomaly-detection/pull/77/commits/4cf9c7a24337dd27b29bbfc77596091c6281778d

**Review by ohltyler — APPROVED** [review]
LGTM

**Review by spbjss — APPROVED** [review]
looks good to me.

**Pull request closed without merging**